lib/es6/labeler.js

```diff
@@ -0,0 +1,84 @@
+import octonode from 'octonode';
+import Q from 'q';
+import _ from 'lodash';
+import { getLogger } from 'appium-logger';
+import { repoForIssue } from './utils';
+
+let labelerSingleton;
+
+class IssueLabeler {
+  constructor (repoConfig:Object):void {
+    this.bot = repoConfig.bot;
+    this.repos = repoConfig.repos;
+    this.logger = getLogger();
+  }
+
+  repoStr (repo:Object):string {
+    return `${repo.user}/${repo.repo}`;
+  }
+
+  async labelIssue (data) {
+    let [ourRepo, theirUser, theirRepo] = repoForIssue(data, this.repos);
+    let sender = data.sender.login;
+    this.validLabels = ourRepo.validLabels;
+    if (!this.shouldLabelIssue(ourRepo, data)) {
+      this.logger.info("No label request. Doing nothing.");
+      return;
+    }
+    this.logger.info(`@${sender} wants to label #${data.issue.number}`);
+
+    if (_.contains(ourRepo.triagers, sender)) {
+      this.logger.info('Permission granted.');
+      await this.performLabeling(this.repoStr(ourRepo), data.issue.number);
+    } else {
+      this.logger.info('Permission denied.');
+    }
+  }
+
+  shouldLabelIssue (ourRepo:Object, data:Object):boolean {
+    this.labels = [];
+    let exp = new RegExp(`@${this.bot} please label:{0,1}(.*).`, 'i')
```

Comment:
This is as a safeguard in case there's more text after the label request. Perhaps we could look for a period, new line or end of line?

```
@triager please label x, y, z

more comments, q, r, s
```